src/main/scala/me/jeffmay/neo4j/client/ws/WSNeo4jClient.scala

```diff
@@ -0,0 +1,208 @@
+    val request = http(s"/user/${config.credentials.username}/password")
+    request.postAndCheckStatus(Some(Json.obj("password" -> newPassword)), Set(200))
+      .map(_ => ())
+  }
+
+  override def openTxn(statements: Seq[CypherStatement]): Future[OpenedTxnResponse] = {
+    // TODO: Allow configuration per-request
+    val jsonBody = Json.toJson(RawNeo4jCypherRequest(statements, includeStats = Some(config.includeStatsByDefault)))
+    val request = http("/db/data/transaction")
+    whenInitialized {
+      request.postAndCheckStatus(Some(jsonBody)).flatMap { resp =>
+        // TODO: Handle version / json format errors with common recovery code
+        val respBody = resp.json.as[RawTxnResponse]
+        if (respBody.isSuccessful) {
+          Future.successful(respBody.asOpenTxnResponse.get)
+        }
+        else {
+          Future.failed(statusCodeException(request, Some(jsonBody), resp, respBody.errors))
+        }
+      }
+    }
+  }
+
+  override def openAndCommitTxn(statements: Seq[CypherStatement]): Future[CommittedTxnResponse] = {
+    // TODO: Allow configuration per-request
+    val jsonBody = Json.toJson(RawNeo4jCypherRequest(statements, includeStats = Some(config.includeStatsByDefault)))
+    val request = http("/db/data/transaction/commit")
+    whenInitialized {
+      request.postAndCheckStatus(Some(jsonBody)).flatMap { resp =>
+        // TODO: Handle version / json format errors with common recovery code
+        val respBody = resp.json.as[RawTxnResponse]
+        if (respBody.isSuccessful) {
+          Future.successful(respBody.asCommitTxnResponse.get)
+        }
+        else {
+          Future.failed(statusCodeException(request, Some(jsonBody), resp, respBody.errors))
+        }
+      }
+    }
+  }
+
+  override def commitTxn(ref: TxnRef, statements: Seq[CypherStatement]): Future[CommittedTxnResponse] = {
+    // TODO: Allow configuration per-request
+    val jsonBody = Json.toJson(RawNeo4jCypherRequest(statements, includeStats = Some(config.includeStatsByDefault)))
+    val request = http("/db/data/transaction")
+    whenInitialized {
+      request.postAndCheckStatus(Some(jsonBody)).flatMap { resp =>
+        // TODO: Handle version / json format errors with common recovery code
+        val respBody = resp.json.as[RawTxnResponse]
+        if (respBody.isSuccessful) {
+          Future.successful(respBody.asCommitTxnResponse.get)
+        }
+        else {
+          Future.failed(statusCodeException(request, Some(jsonBody), resp, respBody.errors))
+        }
+      }
+    }
+  }
+}
+
+object WSNeo4jClient {
+
+  // TODO: Replace this with an actor to manage throttling / circuit breaker / backoff retry on unauthorized or connection issues
+
+  @volatile private var lastSuccessfulRequestWithCreds: Option[Neo4jBasicAuth] = None
+
+  // this is a very naive implementation of recovery that should evolve into a more robust / non-blocking solution
```

Comment:
I'm thinking of removing it. In live code, you should have a pre-setup database. If you don't have one, you can use a helper method to initial the db. The tricky part is doing the setup before and after tests. For that I have some test helper code that I use for testing Neo4j. So I'm thinking I can just expose that in another module in sbt that you can depend on for running tests with these hooks.